src/frontend/staff-portal/src/app/components/ticket-info/ticket-info.component.ts

```diff
@@ -1,147 +1,575 @@
+      this.form.get('violationTicket').get('driversLicenceNumber').setValidators([Validators.required, Validators.minLength(7), Validators.maxLength(9), Validators.pattern(/^(\d{7}|\d{8}|\d{9})$/)]);
+      this.form.get('violationTicket').get('driversLicenceNumber').updateValueAndValidity();
+      this.form.updateValueAndValidity();
+
+    } else {
+      this.form.get('violationTicket').get('driversLicenceNumber').setValidators(Validators.required);
+      this.form.get('violationTicket').get('driversLicenceNumber').updateValueAndValidity();
+      this.form.updateValueAndValidity();
+    }
+  }
+
+  // change validators on drivers licence number in notice of dispute when changing province / state
+  public onNoticeOfDisputeDLProvinceChange(province: string) {
+    if (province == 'British Columbia') {
+      this.form.get('driversLicenceNumber').setValidators([Validators.required, Validators.minLength(7), Validators.maxLength(9), Validators.pattern(/^(\d{7}|\d{8}|\d{9})$/)]);
+      this.form.get('driversLicenceNumber').updateValueAndValidity();
+      this.form.updateValueAndValidity();
+
+    } else {
+      this.form.get('driversLicenceNumber').setValidators(Validators.required);
+      this.form.get('driversLicenceNumber').updateValueAndValidity();
+      this.form.updateValueAndValidity();
+    }
+  }
+  onKeyPressNumbers(event: any, BCOnly: boolean) {
+    var charCode = (event.which) ? event.which : event.keyCode;
+    // Only Numbers 0-9
+    if ((charCode < 48 || charCode > 57) && BCOnly) {
+      event.preventDefault();
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  public onSubmitViolationTicket(): void {
+
+    // We are only sending the violation Ticket fields so update a local copy of lastUpdatedDispute
+    // with violation Ticket form fields only that were changed
+    let putDispute = this.lastUpdatedDispute;
+
+    putDispute.violationTicket.ticketNumber = this.form.get('violationTicket').get('ticketNumber').value;
+    putDispute.violationTicket.surname = this.form.get('violationTicket').get('surname').value;
+    putDispute.violationTicket.givenNames = this.form.get('violationTicket').get('givenNames').value;
+    putDispute.violationTicket.driversLicenceNumber = this.form.get('violationTicket').get('driversLicenceNumber').value;
+    putDispute.violationTicket.driversLicenceProvince = this.form.get('violationTicket').get('driversLicenceProvince').value;
+
+    // provincial court hearing is returned by the GET at the level of notice of dispute but is to be saved with
+    // violation Ticket fields not notice of dispute fields
+    putDispute.provincialCourtHearingLocation = this.form.get('violationTicket').get('provincialCourtHearingLocation').value;
+
+    // reconstruct issued date as string from violation date and violation time format yyyy-mm-ddThh:mm
+    putDispute.violationTicket.issuedDate =
+      this.form.get('violationTicket').get('violationDate').value +
```

Comment:
suggest to use angular default datepipe to convert the date
this.datePipe.transform(this.form.get('violationTicket').get('violationDate').value + " " + this.form.get('violationTicket').get('violationTime').value, "YYYY-MM-ddTHH:mm:ss");